src/librepgp/stream-parse.cpp

```diff
@@ -1880,7 +1881,6 @@ init_encrypted_src(pgp_processing_ctx_t *ctx, pgp_source_t *src, pgp_source_t *r
     }
 
     src->read = param->aead ? encrypted_src_read_aead : encrypted_src_read_cfb;
-    src->close = encrypted_src_close;
     src->finish = encrypted_src_finish;
     src->type = PGP_STREAM_ENCRYPTED;
```

Comment:
These ones should be also moved, so stream is fully initialized before possible error exit.

Reply:
moving these lines makes 3 tests fail :)

```
--- a/src/librepgp/stream-parse.cpp
+++ b/src/librepgp/stream-parse.cpp
@@ -1872,7 +1872,11 @@ init_encrypted_src(pgp_processing_ctx_t *ctx, pgp_source_t *src, pgp_source_t *r
     }
     param = (pgp_source_encrypted_param_t *) src->param;
     param->pkt.readsrc = readsrc;
+
     src->close = encrypted_src_close;
+    src->read = param->aead ? encrypted_src_read_aead : encrypted_src_read_cfb;
+    src->finish = encrypted_src_finish;
+    src->type = PGP_STREAM_ENCRYPTED;
 
     /* Read the packet-related information */
     errcode = encrypted_read_packet_data(param);
@@ -1880,10 +1884,6 @@ init_encrypted_src(pgp_processing_ctx_t *ctx, pgp_source_t *src, pgp_source_t *r
         goto finish;
     }
 
-    src->read = param->aead ? encrypted_src_read_aead : encrypted_src_read_cfb;
-    src->finish = encrypted_src_finish;
-    src->type = PGP_STREAM_ENCRYPTED;
-
     /* Obtaining the symmetric key */
     have_key = false;
 

```

Reply:
I guess `param->aead` is initialized by encrypted_read_packet_data() so this line
`src->read = param->aead ? encrypted_src_read_aead : encrypted_src_read_cfb;`
should be left on its place

Reply:
@antonsviridenko Correct, this one should be set after the `encrypted_read_packet_data()`.